ksql-common/src/test/java/io/confluent/ksql/util/SchemaUtilTest.java

```diff
@@ -44,4 +63,231 @@ public void shouldCreateCorrectAvroSchema() throws IOException {
   }
 
+  @Test
+  public void shouldGetTheCorrectJavaTypeForBoolean() {
+    Schema schema = Schema.BOOLEAN_SCHEMA;
+    Class javaClass = SchemaUtil.getJavaType(schema);
+    assertThat(javaClass, equalTo(Boolean.class));
+  }
+
+  @Test
+  public void shouldGetTheCorrectJavaTypeForInt() {
+    Schema schema = Schema.INT32_SCHEMA;
+    Class javaClass = SchemaUtil.getJavaType(schema);
+    assertThat(javaClass, equalTo(Integer.class));
+  }
+
+  @Test
+  public void shouldGetTheCorrectJavaTypeForLong() {
+    Schema schema = Schema.INT64_SCHEMA;
+    Class javaClass = SchemaUtil.getJavaType(schema);
+    assertThat(javaClass, equalTo(Long.class));
+  }
+
+  @Test
+  public void shouldGetTheCorrectJavaTypeForDouble() {
+    Schema schema = Schema.FLOAT64_SCHEMA;
+    Class javaClass = SchemaUtil.getJavaType(schema);
+    assertThat(javaClass, equalTo(Double.class));
+  }
+
+  @Test
+  public void shouldGetTheCorrectJavaTypeForString() {
+    Schema schema = Schema.STRING_SCHEMA;
+    Class javaClass = SchemaUtil.getJavaType(schema);
+    assertThat(javaClass, equalTo(String.class));
+  }
+
+  @Test
+  public void shouldGetTheCorrectJavaTypeForArray() {
+    Schema schema = SchemaBuilder.array(Schema.FLOAT64_SCHEMA);
+    Class javaClass = SchemaUtil.getJavaType(schema);
+    assertThat(javaClass, equalTo(new Double[]{}.getClass()));
+  }
+
+  @Test
+  public void shouldGetTheCorrectJavaTypeForMap() {
+    Schema schema = SchemaBuilder.map(Schema.STRING_SCHEMA, Schema.FLOAT64_SCHEMA);
+    Class javaClass = SchemaUtil.getJavaType(schema);
+    assertThat(javaClass, equalTo(HashMap.class));
+  }
+
+  @Test
+  public void shouldFailForCorrectJavaType() {
+
+    try {
+      Class class8 = SchemaUtil.getJavaType(Schema.BYTES_SCHEMA);
+      Assert.fail();
+    } catch (KsqlException ksqlException) {
+      assertThat("Invalid type retured.",ksqlException.getMessage().equals("Type is not "
+                                                                          + "supported: "
+                                                                   + "BYTES"));
+      return;
+    }
+  }
+
+  @Test
+  public void shouldGetTheCorrectFieldName() {
+    Optional<Field> field = SchemaUtil.getFieldByName(schema, "orderid".toUpperCase());
+    Assert.assertTrue(field.isPresent());
+    assertThat("", field.get().schema() == Schema.INT64_SCHEMA);
+    assertThat("", field.get().name().equalsIgnoreCase("orderid"));
+
+    Optional<Field> field1 = SchemaUtil.getFieldByName(schema, "orderid");
+    Assert.assertFalse(field1.isPresent());
+  }
+
+  @Test
+  public void shouldGetTheCorrectSchemaForBoolean() {
+    Schema schema = SchemaUtil.getTypeSchema("BOOLEAN");
+    assertThat("", schema == Schema.BOOLEAN_SCHEMA);
```

Comment:
nit: we could use assertThat(schema, sameInstance(Schema.BOOLEAN_SCHEMA)). This way we would get a nicer error message on failure.

Reply:
yep, same elsewhere in this test

Reply:
Good point, changed accordingly.